### PR TITLE
Decoupled reference flow (part3)

### PR DESCRIPTION
**Pull request by davidgisbey**
## Context

Previous PRs are:
#3013 - Adds the new start and type page
#3016 - Migration to allow the references email address column to be NULL
#3022 - Breaks the new flow into a separate controller
#3035 - Up to adding a referees email 

This PR adds the email address view, controller actions, form and routes. The description page onwards is out of scope.

## Changes proposed in this pull request

image

- With no value 

image

- With the incorrect format 

image

- With a duplicate email

image

## Guidance to review

I'm not overly happy with the validations. I'd love some input on them if anyone has any ideas.

## Link to Trello card

https://trello.com/c/npGvdYqK/2213-%F0%9F%92%94-dev-request-a-referee-flow

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

File: app/views/candidate_interface/decoupled_references/email_address/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+            <strong class="govuk-tag govuk-phase-banner__content__tag app-environment-tag govuk-tag--purple">
+              sandbox feature
+            </strong>
+          </p>
+          <p class="govuk-body govuk-!-margin-bottom-0">
+            Enter <code>refbot1@example.com</code> and <code>refbot2@example.com</code> as your referee email addresses to have the references automatically completed
+          </p>
+        </div>
+      <% end %>
+      <%= f.govuk_text_field :email_address, type: 'email', label: { text: '' }, hint_text: 'In most cases, this should be a work address.' %>
```
Comment (tvararu) [inline app/views/candidate_interface/decoupled_references/email_address/new.html.erb:29]: Does the label need to be empty? Can it be the <h1> above somehow?

File: config/routes.rb [hunk 2]
```diff
@@ -394,10 +392,18 @@
       end
 
       scope '/references' do
-        get '/start' => 'decoupled_references#start', as: :decoupled_references_start
+        get '/start' => 'decoupled_references/base#start', as: :decoupled_references_start
+
+        get '/type' => 'decoupled_references/type#new', as: :decoupled_references_new_type
+        post '/update-type' => 'decoupled_references/type#create', as: :decoupled_references_create_type
```
Comment (malcolmbaig) [inline config/routes.rb:398]: This is something we can circle back to if we're low on time, but some suggestions about the routing:

```
      get '/type' => 'decoupled_references/type#new', as: :decoupled_references_type
      post '/type' => 'decoupled_references/type#create'
```
Means you only need the one path helper and can use its path for both GETs and POSTs.

File: config/routes.rb [hunk 3]
```diff
@@ -394,10 +392,18 @@
...
+
+        get '/name/:id' => 'decoupled_references/name#new', as: :decoupled_references_new_name
+        post '/name/:id' => 'decoupled_references/name#create', as: :decoupled_references_create_name
```
Comment (malcolmbaig) [inline config/routes.rb:401]: Making the `post` a `patch` means we use the correct HTTP verb to get here, and provides another hint that we're updating an existing resource at this endpoint. Also, an `edit/update` action pair is probably more accurate than `new/create`.

Comment (davidgisbey) [reply config/routes.rb:401]: Okay cool. We need additional actions to go back and edit/update their original choice, then redirect to the unsubmitted review page. I think for now it's probs best to leave it at this until the flow is working. Then I'll come back and make the requested changes. I've added them to the card.

File: config/routes.rb (hunk 2 above)
Comment (davidgisbey) [reply config/routes.rb]: Same as below 👍

**Review by malcolmbaig — APPROVED** [review]

File: app/views/candidate_interface/decoupled_references/email_address/new.html.erb (hunk 1 above)
Comment (davidgisbey) [reply app/views/candidate_interface/decoupled_references/email_address/new.html.erb]: Done 👍

**Pull request merged**